src/main/python/modelers/regression_models.py

```diff
@@ -0,0 +1,122 @@
+import numpy as np
+import pandas as pd
+from argparse import Namespace
+from sklearn.model_selection import train_test_split
+from sklearn.linear_model import LinearRegression
+from sklearn.neural_network import MLPRegressor
+from sklearn.ensemble import RandomForestRegressor
+from sklearn.metrics import r2_score
+from sklearn.metrics import mean_squared_error
+
+
+class TweetsModeler:
+    def __init__(self, args):
+        self.dfWithFeatures = args.dfWithFeatures
+        self.featureColumnIndexes = args.featureColumnIndexes
+        self.points = args.points
+        self.regressionMethod = args.regressionMethod
+        self.regResultsDF = pd.DataFrame()
+        self.resultsDict = dict()
+
+    @classmethod
```

Comment:
If all you're doing with the class is calling methods without ever creating the object and manipulating class variables, there's not too much of a point of making it a class. This is unique to scripting type languages that are also OOP, and one of the things that I enjoy most about developing in Python versus Java. 
I think you may have done this to prevent passing so many arguments to the different model methods, which is also ideal in OOP, but the real point of classes is that they contain information for other classes to use and manipulate since the data should be private to the class. This idea is explained in a fantastic book called Clean Code under a chapter about Hybrid classes and the difference between data structures and objects. In this case, you are more creating a data structure than a class because all you are doing is storing the data and not manipulating it. While you are using the data in the class to return a new result, you don't really need a class for it. Here is a link that it explains it well https://stackoverflow.com/a/21064263/7658632
In his answer, he mentions the use of static methods (similar to class methods, difference here: https://www.tutorialspoint.com/class-method-vs-static-method-in-python#:~:text=The%20static%20method%20does%20not,the%20state%20of%20that%20class.) like what you have, but he only mentions these static methods because in Java everything needs to be an object, which is typically a class. In python, if you can avoid creating an object, you should because it increases the resources used by the machine.
I am fine with you keeping it this, but figured I'd share some experience as a developer for you. 
I JUST NOTICED!!!
I think you should put the print_results method in main into this class instead. That would then make this more of a class, although it still lacks a lot elements that would make it a class.
If you do that, I'll be even happier but you don't have to. Just let me know what you choose.